web/packages/teleport/src/Teleport.tsx

```diff
@@ -101,15 +103,17 @@ const Teleport: React.FC<Props> = props => {
                   <Route path={cfg.routes.root}>
                     <Authenticated>
                       <UserContextProvider>
-                        <TeleportContextProvider ctx={ctx}>
-                          <Switch>
-                            <Route
-                              path={cfg.routes.appLauncher}
-                              component={AppLauncher}
-                            />
-                            <Route>{createPrivateRoutes()}</Route>
-                          </Switch>
-                        </TeleportContextProvider>
+                        <ToastNotificationProvider>
+                          <TeleportContextProvider ctx={ctx}>
+                            <Switch>
+                              <Route
+                                path={cfg.routes.appLauncher}
+                                component={AppLauncher}
+                              />
+                              <Route>{createPrivateRoutes()}</Route>
+                            </Switch>
+                          </TeleportContextProvider>
+                        </ToastNotificationProvider>
```

Comment:
i had to put the toast provider here because of how we handle user invites for starting up cloud cluster. changes found here: https://github.com/gravitational/teleport.e/pull/6908

This is the flow:
- start up cloud cluster
- create user/password
- before going to the dashboard `Main.tsx`, it asks you to invite collaborators
- afterwards, the `Main.tsx` renders, and then we render the toast notifying users if invitation succeeded or not

Reply:
Doesn't it mean that we'll have two nested toast notification providers, and depending on route, we'll be using one or another?

Reply:
hrm, i wasn't entirely sure what you meant. there is only one `ToastNotificationProvider` defined atm 🤔

Reply:
Oops, I must have either confused `TestNotifications` with `TestNotificationProvider` or confused fragments of `Main.tsx` with `Main.test.tsx`. Never mind. :)